<?xml version="1.0"?>
<robot name="materials">

  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>

  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  
  <material name="blue">
    <color rgba="0 0 1 1"/>
  </material>
  
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>


  <link name="base_link">
    <visual>
      <geometry>
        <box size="6 1 1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
      <material name="red"/>
    </visual>
    
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
     <geometry>
        <box size="6 1 1"/>
     </geometry>
   </collision>
  </link>
  
  <link name="1_link">
    <visual>
      <geometry>
        <box size="1 1 6"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 3.0"/>
      <material name="green"/>
    </visual>
    
    <collision>
      <origin rpy="0 0 0" xyz="0 0 3.0"/>
     <geometry>
        <box size="1 1 6"/>
     </geometry>
   </collision>
  </link>
  
  <joint name="base_to_1_link" type="prismatic">
    <parent link="base_link"/>
    <child link="1_link"/>
	<limit effort="10.0" lower="0" upper="5" velocity="0.1"/>
    <origin xyz="-2.5 0 1"/>
  </joint>
  
  <link name="2_link">
    <visual>
      <geometry>
        <box size="1 5.5 1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 2.25 0.5"/>
      <material name="blue"/>
    </visual>
    
    <collision>
      <origin rpy="0 0 0" xyz="0 2.25 0.5"/>
     <geometry>
        <box size="1 5.5 1"/>
     </geometry>
   </collision>
  </link>
  
  <joint name="1_to_2_link" type="prismatic">
	<axis xyz="0 1 0"/>
    <parent link="1_link"/>
    <child link="2_link"/>
	<limit effort="10.0" lower="-4.5" upper="0" velocity="0.1"/>
    <origin xyz="0 0 6"/>
  </joint>
  
  <link name="3_joint_link">
    <visual>
      <geometry>
	<cylinder radius="0.5" length="2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
      <material name="black"/>
    </visual>
  </link>
  
  <joint name="2_to_3_joint_link" type="revolute">
	<axis xyz="0 0 1"/>
    <parent link="2_link"/>
    <child link="3_joint_link"/>
	<limit effort="10.0" lower="-3.14" upper="3.14" velocity="0.1"/>
    <origin xyz="0 5.0 0.5"/>
  </joint>
  
  <link name="3_link">
    <visual>
      <geometry>
        <box size="1 3 1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
      <material name="black"/>
    </visual>
    
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
     <geometry>
        <box size="1 3 1"/>
     </geometry>
   </collision>
  </link>
  
  <joint name="3_joint_to_3_link" type="fixed">
    <parent link="3_joint_link"/>
    <child link="3_link"/>
    <origin xyz="0 0 0.5"/>
  </joint>

</robot>
